sysvar/src/lib.rs

```diff
@@ -174,11 +177,43 @@ macro_rules! impl_sysvar_get {
 
             match result {
                 $crate::__private::SUCCESS => Ok(var),
-                // Unexpected errors are folded into `UnsupportedSysvar`.
                 _ => Err($crate::__private::ProgramError::UnsupportedSysvar),
             }
         }
     };
+    // Variant for sysvars with padding at the end. Loads bincode-serialized data
+    // (size - padding bytes) and zeros the padding to avoid undefined behavior.
+    // Only supports sysvars where padding is at the end of the layout. Caller
+    // must supply the correct number of padding bytes.
+    ($sysvar_id:expr, $padding:literal) => {
+        fn get() -> Result<Self, $crate::__private::ProgramError> {
+            let mut var = core::mem::MaybeUninit::<Self>::uninit();
+            let var_addr = var.as_mut_ptr() as *mut u8;
+            let length = core::mem::size_of::<Self>().saturating_sub($padding);
+            let sysvar_id_ptr = (&$sysvar_id) as *const _ as *const u8;
+            // SAFETY: The allocation is valid for `size_of::<Self>()`. We load
+            // `(size - padding)` bytes from the syscall, which matches bincode
+            // serialization. The remaining `padding` bytes are then zeroed.
+            let result =
+                unsafe { $crate::get_sysvar_unchecked(var_addr, sysvar_id_ptr, 0, length as u64) };
+            match result {
+                Ok(()) => {
+                    // SAFETY: All bytes now initialized: syscall filled data
+                    // bytes and we zeroed padding.
+                    unsafe {
+                        var_addr.add(length).write_bytes(0, $padding);
+                        Ok(var.assume_init())
+                    }
```

Comment:
nit: The Pinocchio version does the zero-ing before the syscall, should we do the same in this implementation?